charm/kafka/lib/charms/layer/kafka.py

```diff
@@ -127,6 +127,13 @@ def install(self, zk_units=[], log_dir='logs'):
             context=context
         )
 
+        log4j_file = os.path.join(KAFKA_SNAP_DATA, 'log4j.properties')
+        if config.get('log4j_properties'):
+            with open(log4j_file, 'w') as f:
+                print(config['log4j_properties'], file=f)
```

Comment:
Out of curiosity: why `print(file=...)` insteaf of simply `f.write`?

Reply:
f.write expects a `bytes` argument, but the config value is a utf8 `str`. Alternatively I could have explicitly encoded the `str` to a `bytes` as utf-8.